packages/runtime/runtime-utils/src/summarizerNode.ts

```diff
@@ -146,6 +146,14 @@ function decodeSummary(snapshot: ISnapshotTree, logger: Pick<ITelemetryLogger, "
                     let outstandingOps: ISequencedDocumentMessage[] = [];
                     for (const opsBlob of opsBlobs) {
                         const newOutstandingOps = await readAndParseBlob<ISequencedDocumentMessage[]>(opsBlob);
+                        if (outstandingOps.length > 0 && newOutstandingOps.length > 0) {
+                            const latestSeq = outstandingOps[outstandingOps.length - 1].sequenceNumber;
+                            const newEarliestSeq = newOutstandingOps[0].sequenceNumber;
+                            assert(
+                                latestSeq < newEarliestSeq,
+                                `latestSeq exceeds newEarliestSeq in decodeSummary: ${latestSeq} >= ${newEarliestSeq}`,
+                            );
+                        }
```

Comment:
Added this assert which will throw for broken docs while loading them.
I think at this point, it could be possible to correct the data by removing duplicate ops, do you think it would be valuable to add that code - at least temporarily?
@vladsud